tests/test_load_yaml_dict.py

```diff
@@ -65,51 +69,114 @@ def test_get_env_name(mocker, tmpdir, cmd_line_name) -> None:
 
     from conda_devenv.devenv import get_env_name
 
-    name = get_env_name(args, filename, None)
+    name = get_env_name(args, yaml.safe_load(filename.read()))
     if cmd_line_name:
         assert name == "foo"
     else:
         assert name == "bar"
 
 
 def test_is_included_var(datadir) -> None:
-    import textwrap
-
     a_env_file = datadir / "a.devenv.yml"
     a_env_file.write_text(
-        str(
-            textwrap.dedent(
-                """\
-        name: a
-        includes:
-          - {{root}}/b.devenv.yml
-        environment:
-          VARIABLE: value_a
-          IS_A_INCLUDED: {{is_included}}
-    """
-            )
+        textwrap.dedent(
+            """
+            name: a
+            includes:
+              - {{root}}/b.devenv.yml
+            environment:
+              VARIABLE: value_a
+              IS_A_INCLUDED: {{is_included}}
+            """
         )
     )
     b_env_file = datadir / "b.devenv.yml"
     b_env_file.write_text(
-        str(
-            textwrap.dedent(
-                """\
-        name: b
-        environment:
-          {% if not is_included %}
-          VARIABLE: value_b
-          {% endif %}
-          IS_B_INCLUDED: {{is_included}}
-    """
-            )
+        textwrap.dedent(
+            """
+            name: b
+            environment:
+              {% if not is_included %}
+              VARIABLE: value_b
+              {% endif %}
+              IS_B_INCLUDED: {{is_included}}
+            """
+        )
+    )
+
+    conda_env = load_yaml_dict(a_env_file)
+    assert conda_env == {
+        "name": "a",
+        "environment": {
+            "IS_A_INCLUDED": False,
+            "IS_B_INCLUDED": True,
+            "VARIABLE": "value_a",
+        },
+    }
+
+
+def test_root_overrides_channels(tmp_path) -> None:
+    a_fn = tmp_path / "a.devenv.yml"
+    a_fn.write_text(
+        textwrap.dedent(
+            """
+            name: a
+            channels:
+            - a_channel
+            """
+        )
+    )
+
+    b_fn = tmp_path / "b.devenv.yml"
+    b_fn.write_text(
+        textwrap.dedent(
+            """
+            name: b
+            includes:
+              - {{ root }}/a.devenv.yml
+            channels:
+            - b1_channel
+            - b2_channel
+            """
+        )
+    )
+
+    assert load_yaml_dict(b_fn) == {
+        "name": "b",
+        "channels": ["b1_channel", "b2_channel"],
+        "environment": {},
+    }
```

Comment:
I think so, the downstream/leaf `environment.devenv.yml` file should have total control to override the channels entirely, given its file will be used in the end for the final solving... but I'm not 100% sure, seems safer this way, let me know what you think.

Reply:
OK. I was just "thing out loud".

I remember that given we that offer the template variable `root` we decided to avoid using `root` elsewhere =) but by "" I think you mean root.

```
def load_yaml_dict(filename: Path) -> tuple[YAMLData, dict]:
    ...
    rendered_contents = render_jinja(contents, filename, is_included=False)
    root_yaml = yaml.safe_load(rendered_contents)
    all_yaml_dicts = handle_includes(filename, root_yaml)
    ...
    merged_dict = merge(all_yaml_dicts.values())
    ...
```

Reply:
Ahh you mean the name of the test. Definitely, fixed it.